readme.md

```diff
@@ -104,20 +151,48 @@ Whether to treat all messages as turned off initially (`boolean`, default:
 
 ###### `options.enable`
 
-List of `ruleId`s to initially turn on if `reset: true` (`Array.<string>`,
+List of `ruleId`s to initially turn on if `reset: true` (`Array<string>`,
 optional).
-By default (`reset: false`), all rules are turned on.
+All rules are turned on by default (`reset: false`).
 
 ###### `options.disable`
 
-List of `ruleId`s to turn on if `reset: false` (`Array.<string>`, optional).
+List of `ruleId`s to turn on if `reset: false` (`Array<string>`, optional).
 
 ###### `options.sources`
 
 Sources that can be controlled with `name` markers (`string` or
-`Array.<string>`, default: `options.name`).
+`Array<string>`, default: `options.name`).
 
-### Markers
+<!--Old name of section-->
+
+<a name="markers"></a>
+
+## Syntax
+
+This plugin looks for comments in markdown (MDX is also supported).
+If the first word in those comments does not match `options.name`, the comment
+is skipped.
+The second word is expected to be `disable`, `enable`, or `ignore`.
+Further words are rule identifiers of messages which are configurated.
+
+In EBNF, the grammar looks as follows:
+
+<pre><code class=language-ebnf><a id=s-marker href=#s-marker>marker</a> ::= <a href=#s-identifier>identifier</a> <a href=#s-whitespace>whitespace</a>+ <a href=#s-keyword>keyword</a> <a href=#s-ruleidentifiers>ruleIdentifiers</a>?
+<a id=s-identifier href=#s-identifier>identifier</a> ::= <a href=#s-word>word</a>+ /* restriction: must match `options.name` */
+<a id=s-keyword href=#s-keyword>keyword</a> ::= 'enable' | 'disable' | 'ignore'
+<a id=s-ruleidentifiers href=#s-ruleidentifiers>ruleIdentifiers</a> ::= <a href=#s-word>word</a>+ (<a href=#s-whitespace>whitespace</a>+ <a href=#s-word>word</a>+)*
+<a id=s-whitespace href=#s-whitespace>whitespace</a> ::= ' ' | '\t' | '\r' | '\n' | '\f'
+<a id=s-word href=#s-word>word</a> ::= <a href=#s-letter>letter</a> | <a href=#s-digit>digit</a> | <a href=#s-punctuation>punctuation</a>
+<a id=s-letter href=#s-letter>letter</a> ::= <a href=#s-letterlowercase>letterLowercase</a> | <a href=#s-letteruppercase>letterUppercase</a>
+<a id=s-punctuation href=#s-punctuation>punctuation</a> ::= '-' | '_'
+<a id=s-letterlowercase href=#s-letterlowercase>letterLowercase</a> ::= 'a' | 'b' | 'c' | 'd' | 'e' | 'f' | 'g' | 'h' | 'i' | 'j' | 'k' | 'l' | 'm' | 'n' | 'o' | 'p' | 'q' | 'r' | 's' | 't' | 'u' | 'v' | 'w' | 'x' | 'y' | 'z'
+<a id=s-letteruppercase href=#s-letteruppercase>letterUppercase</a> ::= 'A' | 'B' | 'C' | 'D' | 'E' | 'F' | 'G' | 'H' | 'I' | 'J' | 'K' | 'L' | 'M' | 'N' | 'O' | 'P' | 'Q' | 'R' | 'S' | 'T' | 'U' | 'V' | 'W' | 'X' | 'Y' | 'Z'
+<a id=s-digit href=#s-digit>digit</a> ::= '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9'
+</code></pre>
```

Comment:
Interesting, does GH linguist not recognize EBNF notation?

Reply:
It does, but I doesn't link identifiers. This way (more useful in mdx jsx and such) you can link things together!

Reply:
Oh? How to you see the linking together working?
What is the value add in linking to a specific line in a 10 line example versus linking to the EBNF section and mentioning the line/group name?

Reply:
See the rendered preview, or see [`remark-gemoji`](https://github.com/remarkjs/remark-gemoji#syntax) or [`micromark-extension-mdx-jsx`](https://github.com/micromark/micromark-extension-mdx-jsx#syntax)!

Reply:
Right, I get how it works.

Why do we need to go out of the way to add extra html ids to specific identifiers?
I don't see those being linked to in this document outside the codeblock, and haven't seen other readmes linking to these.

Reply:
I’m not sure if this is better than using a regular markdown code block. Using a reference / pointer / anchor isn’t unique to EBNF, i.e. the same could be applied to JavaScript identifiers, YAML anchors, or JSON pointers, etc.

I’d leave this up to the markdown renderer (GitHub in this case). Also I’m not sure if npmjs suppors it, which will also render the readme.

Reply:
It’s true that it is a bit of work to add them, however, the larger this stuff becomes, the more useful it becomes to cross-link between things.

> haven't seen other readmes linking to these.

Largely because they didn’t exist ':)

> I’m not sure if npmjs suppors it

They should!

---

Note that I want to automate this in the future. See https://github.com/remarkjs/remark-gemoji/pull/10#issuecomment-963050982 for an example. I’d like to have clean EBNF in code, rendered EBNF on GH/etc, and a diagram.

Reply:
If it’s automated and both GitHub and npm support it, it’s ok with me. Although I do think it’s a bit unfortunate this comes at the cost of readability of the raw markdown file.